doc/api/http.md

```diff
@@ -2440,6 +2440,69 @@ events will be emitted in the following order:
 Setting the `timeout` option or using the `setTimeout()` function will
 not abort the request or do anything besides add a `'timeout'` event.
 
+## `http.validateHeaderName(name)`
+
+* `name` {string}
+
+Performs the low-level validations on the provided `name` that are done when
+`res.setHeader(name, value)` is called.
+
+Passing illegal value as `name` will result in a [`TypeError`][] being thrown,
+identified by `code: 'ERR_INVALID_HTTP_TOKEN'`.
+
+You don't have to use it to validate each header you pass to `res.setHeader` -
+it is used there anyway.
+
+Example:
+```js
+const { validateHeaderName } = require('http');
+
+try {
+    validateHeaderName('')
+} catch(err) {
+    err instanceof TypeError // --> true
```

Comment:
I’m not sure we should document this, partly because it seems wrong to me to use `TypeError` here.
```suggestion
```